src/ClientHandle.cpp

```diff
@@ -2101,6 +2078,19 @@ bool cClientHandle::CheckBlockInteractionsRate(void)
 
 
 
+bool cClientHandle::IsWithinReach(const Vector3i a_Position) const
+{
+	// Distance from the block's center to the player's eye height.
+	const double Distance = (Vector3d(0.5, 0.5, 0.5) + a_Position - m_Player->GetEyePosition()).SqrLength();
+
+	// _X 2014 - 11 - 25: I've maxed at 5.26 with a Survival client and 5.78 with a Creative client in my tests.
+	return Distance <= (m_Player->IsGameModeCreative() ? 33.4084 : 27.6676);
+}
```

Comment:
I think this ought to be a player function, rather than a clientHandle function

Reply:
Now that you mention it I've had the thought that all of the client event handlers should be moved to player. Client handle does whatever checks it needs and then calls OnRightClicked/OnBlockDigBegin/OnBowCharging/and so on, leaving the player to process it. I don't think having player effectively expose all of its members via getters so that ClientHandle can act on its behalf is very nice.

Reply:
Dunno about moving it right now, can we leave it?

Reply:
Yeah, it's fine if you want to keep it for now.